scrapy/link.py

```diff
@@ -39,7 +37,7 @@ def __init__(
         self.fragment: str = fragment
         self.nofollow: bool = nofollow
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
```

Comment:
Interesting.

Reply:
Yeah. https://docs.astral.sh/ruff/rules/any-eq-ne-annotation/ says "The Python documentation recommends the use of object to "indicate that a value could be any type in a typesafe manner". Any, on the other hand, should be seen as an "escape hatch when you need to mix dynamically and statically typed code"." which makes sense and may be useful in other places too.